tasks/mysql_secure_installation.yml

```diff
@@ -0,0 +1,63 @@
+---
+
+# supported for ansible ver => 2.0
+#- name: Install python-mysqldb for Ansible
+#  package: pkg=python-mysqldb state=present
+
+
+- name: Install python-mysqldb for Ansible
+  apt: name=python-mysqldb state=present
+  when: ansible_distribution == 'Debian' or ansible_distribution == 'Ubuntu'
+
+- name: Install python-mysqldb for Ansible
+  yum: name=MySQL-python state=present
+  when: ansible_os_family == 'RedHat' or ansible_os_family == 'Oracle Linux'
+
+- name: root password is present
+  mysql_user: name=root host={{ item }} password={{ mysql_root_password }} state=present
+  with_items:
+    - '::1'
+    - '127.0.0.1'
+    - 'localhost'
+
+- name: root has .my.cnf
+  template: src=my.cnf.j2 dest=/root/.my.cnf 
```

Comment:
There's a problem with `dest=/root/.my.cnf`. If the user running the role is not root but e.g. becomes root with sudo, login will fail. That is because mysql will look for the .my.cnf in users-home directory, not in the root-directory.
Best way to mitigate this is to let the home-path be defineable with a variable, e.g. `mysql_user_home=/root/`